lua/lualine/init.lua

```diff
@@ -251,19 +251,17 @@ end
 
 local function set_tabline()
   if next(config.tabline) ~= nil then
-    _G.lualine_tabline = tabline
-    vim.o.tabline = '%!v:lua.lualine_tabline()'
+    vim.o.tabline = "%!v:lua.require'lualine'.tabline()"
     vim.o.showtabline = 2
   end
 end
 
 local function set_statusline()
   if next(config.sections) ~= nil or next(config.inactive_sections) ~= nil then
-    _G.lualine_statusline = status_dispatch
-    vim.o.statusline = '%!v:lua.lualine_statusline()'
+    vim.o.statusline = "%!v:lua.require'lualine'.statusline()"
     vim.api.nvim_exec([[
-    autocmd lualine WinLeave,BufLeave * lua vim.wo.statusline=lualine_statusline()
-    autocmd lualine WinEnter,BufEnter * setlocal statusline=%!v:lua.lualine_statusline()
+    autocmd lualine WinLeave,BufLeave * lua vim.wo.statusline=require'lualine'.statusline()
+    autocmd lualine WinEnter,BufEnter * set statusline<
```

Comment:
is the `<` sign something I don't know in lua? I have no idea why that' here. Otherwise lgtm. I have to say, these changes are pretty epic.

I have an idea on how to extend these changes and make it even better but that's out of scope of this pr. I will create an issue with it

Reply:
looks like it silences an echo? Probably better to have
```
 autocmd lualine WinEnter,BufEnter * silent set statusline
```

Reply:
It's a vim feature . set statusline< unsets the winlocal option set through setlocal statusline . Checkout `:h setlocal` it's documented in 2nd and 3rd row .

Reply:
oh right, I learn something about vim everyday 😆 . It's all good then